File: swift/events/ready.swift
import DiscordBM

struct ReadyEvent {
    static func handle(payload: Gateway.Ready, discord: any DiscordClient) async throws {
        let rawUser = try await discord.getOwnUser()
        let user = try rawUser.decode()
        print("Logging in as \(user.username)")
    }
}

File: rust/events/ready.rs
use serenity::async_trait;
use serenity::model::gateway::Ready;
use serenity::prelude::*;

pub struct ReadyEvent;

#[async_trait]
impl EventHandler for ReadyEvent {
    async fn ready(&self, _: Context, ready: Ready) {
        println!("Logged in as {}!", ready.user.name);
    }
}
